ckanext/s3filestore/uploader.py

```diff
@@ -152,12 +189,17 @@ def upload_to_key(self, filepath, upload_file, make_public=False):
                   filepath, self.bucket_name, mime_type)
 
         try:
-            self.get_s3_resource().Object(self.bucket_name, filepath).put(
-                Body=upload_file.read(), ACL='public-read' if make_public else self.acl,
-                ContentType=mime_type)
+            kwargs = {
+                'Body': upload_file.read(), 'ACL': acl,
+                'ContentType': mime_type}
+            if mime_type != 'application/pdf':
+                filename = filepath.split('/')[-1]
+                kwargs['ContentDisposition'] = 'attachment; filename=' + filename
```

Comment:
can you add more metadata please, department name, dataset uuid, dataset nice name, last uploader username, or just the entire metadata we have if possible (no need to restrict)

Reply:
I'll see what's easily available.

Reply:
The package ID is easy enough to add. However, anything else from the package could become outdated if the package changes.

Reply:
maybe add package_last_update_date to the metadata so that it does give when it was last updated.

Reply:
looking at peoples answers on this and reading the api spec, metadata is set at time of upload and can't be altered until a new file is uploaded or a copyfrom command is issued.  

Lets run this past Mark and co on what fields we should drop. 

https://stackoverflow.com/questions/39596987/how-to-update-metadata-of-an-existing-object-in-aws-s3-using-python-boto3